### PR TITLE
App improvements

### PR DESCRIPTION
**Pull request by zigasinko**
https://genialis.atlassian.net/browse/REID-2505
https://genialis.atlassian.net/browse/REID-2504
https://genialis.atlassian.net/browse/REID-2252

### CODE REVIEW

File: src/components/genexpress/modules/clustering/clustering.tsx [hunk 1]
```diff
@@ -59,69 +52,41 @@ export const linkageFunctionOptions: Option<ClusteringLinkageFunction>[] = [
...
     useEffect(() => {
         /* ClusterNode is highlighted if:
          * - all it's children lead to leafs with highlighted genes.
          * - it's gene is among highlighted genes (if node has a gene, it's a leaf node).
          */
-        if (!_.isEmpty(clusterNodes)) {
+        if (
+            clusterByGenesResult != null &&
+            clusterByGenesResult.clusterNodes != null &&
+            !_.isEmpty(clusterByGenesResult?.clusterNodes)
```
Comment (katarina-mursec) [inline src/components/genexpress/modules/clustering/clustering.tsx]: could either optional chaining operator or the not null check be removed?

File: src/components/genexpress/modules/clustering/clustering.tsx [hunk 2]
```diff
@@ -192,8 +151,11 @@ const Clustering = ({
     };
 
     const handleOnHighlightedChanged = (chartHighlightedClusterNodesIds: number[]): void => {
-        const chartHighlightedClusterNodes = clusterNodes.filter((clusterNode) =>
-            chartHighlightedClusterNodesIds.includes(clusterNode.nodeIndex),
+        if (clusterByGenesResult?.clusterNodes == null) {
+            return;
+        }
+        const chartHighlightedClusterNodes = clusterByGenesResult?.clusterNodes?.filter(
```
Comment (katarina-mursec) [inline src/components/genexpress/modules/clustering/clustering.tsx]: is optional chaining operator still needed in addition to the guard clause above?

File: src/components/genexpress/modules/timeSeriesAndGeneSelector/geneSelector/geneSelector/geneSelector.tsx [hunk 3]
```diff
@@ -193,11 +190,9 @@ const GeneSelector = ({
 
     const handleOnChange = (_event: unknown, newValue: Gene[]): void => {
         setValue(newValue);
-        setInputValue('');
+        // setInputValue('');
```
Comment (katarina-mursec) [inline src/components/genexpress/modules/timeSeriesAndGeneSelector/geneSelector/geneSelector/geneSelector.tsx]: can the commented code be removed?

**Review by katarina-mursec — APPROVED** [review]

File: src/components/genexpress/modules/clustering/clustering.tsx (hunk 1 above)
Comment (zigasinko) [reply src/components/genexpress/modules/clustering/clustering.tsx]: Lodash `isEmpty` doesn't include not-null type guard, that's why the first two checks are there. I've removed the redundant chaining operator in the last line.

File: src/components/genexpress/modules/clustering/clustering.tsx (hunk 2 above)
Comment (zigasinko) [reply src/components/genexpress/modules/clustering/clustering.tsx]: It's not.. removed.

**Pull request merged**